libsolidity/interface/CompilerStack.cpp

```diff
@@ -196,6 +197,10 @@ bool CompilerStack::parse()
 				{
 					contract->setDevDocumentation(InterfaceHandler::devDocumentation(*contract));
 					contract->setUserDocumentation(InterfaceHandler::userDocumentation(*contract));
+
+					StaticAnalyzer staticAnalyzer(m_errors);
```

Comment:
I think it is better to use an additional full pass for the static analyzer, i.e. use a new loop over the sources and supply a full source unit to the class, not a single contract. Also, only perform the analysis if `noErrors` is true.